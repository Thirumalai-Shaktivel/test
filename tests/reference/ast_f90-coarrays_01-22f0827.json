{
    "basename": "ast_f90-coarrays_01-22f0827",
    "cmd": "lfortran --show-ast-f90 --no-color {infile}",
    "infile": "tests/coarrays_01.f90",
    "infile_hash": "2f9c4e94855aa3544a5349aca5bef848c4e2dd5a87e0545be958df44",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_f90-coarrays_01-22f0827.stdout",
    "stdout_hash": "0c27d61306973c11939c9f77ff875336037fac7e497203bc8c446e7a",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}