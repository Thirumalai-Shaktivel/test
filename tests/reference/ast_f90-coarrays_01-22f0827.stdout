program coarrays_01
use iso_fortran_env, only: event_type
implicit none
real, dimension(100), codimension[*] :: A
integer :: B[3,*]
integer :: c[*]
real :: D(100,2)[3,*]
real :: E(1,2,3)[1,2,-(1):3,*]
real, allocatable :: F(:)[:]
real, allocatable :: g(:,:,:)[:,:,:]
real, dimension(20), codimension[20,*] :: h
real, codimension[:], allocatable :: z(:,:)
character :: r(20)[20,0:*]
type(event_type) :: ok_to_overwrite[*]
type(event_type), allocatable :: greeting_ready(:)[:]
integer :: i, n
n = 5
allocate(F(n)[*])
allocate(greeting_ready(num_images())[*])
allocate(g(50, 50, 50)[1:2, 1:2, *])
if ((this_image())==(1)) then
    do i = 2, num_images()
        c = (c) + (c[i])
        event post (ok_to_overwrite[i])
    end do
else
    event wait (ok_to_overwrite)
    event post (greeting_ready(this_image())[1])
end if
sync all
c[3] = c[4]
B[1, 2] = B[3, 4]
D(99, 1)[3, 4] = D(1, 2)[1, 2]
D(99, 1)[3, 3] = 5
D(:, 1)[3, 3] = 5
D(:, 1)[3, 3, team=3] = 5
D(:, 1)[3, 3, team=3, stat=i] = 5
D(:, 1)[3, 3, stat=i] = 5
end program coarrays_01
